src/System.Private.CoreLib/src/System/Buffer.CoreCLR.cs

```diff
@@ -37,8 +39,53 @@ internal static unsafe void _ZeroMemory(ref byte b, nuint byteLength)
         [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
         private static extern unsafe void __ZeroMemory(void* b, nuint byteLength);
 
+        // The maximum block size to for __BulkMoveWithWriteBarrier FCall. This is required to avoid GC starvation.
+        private const uint BulkMoveWithWriteBarrierChunk = 0x10000;
```

Comment:
65K - not too big?

Reply:
65k should be less than 50 microseconds. I can make it smaller.